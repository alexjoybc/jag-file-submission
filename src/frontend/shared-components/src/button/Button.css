.normal-blue {
  color: white;
  font-weight: 700;
  font-style: normal;
  background-color: #2d3047;
}

.normal-blue:hover {
  background-color: rgba(43, 91, 164, 1);
  color: white;
}
