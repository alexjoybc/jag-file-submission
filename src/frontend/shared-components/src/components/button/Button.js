import React from "react";
import PropTypes from "prop-types";
import "./Button.css";

export const Button = ({ onClick, label, styling }) => (
  <div>
    <button className={styling} onClick={onClick} type="button">
      {label}
    </button>
  </div>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  styling: PropTypes.string.isRequired
};
