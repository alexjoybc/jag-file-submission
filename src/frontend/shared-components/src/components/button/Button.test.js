import React from "react";
import testBasicSnapshot from "../../TestHelper";

import { Button } from "./Button";

describe("Button Component", () => {
  const onClick = jest.fn();
  const label = "Button label";
  const styling = "normal-blue btn";

  test("Button matches the snapshot", () => {
    const buttonComponent = (
      <Button onClick={onClick} label={label} styling={styling} />
    );

    testBasicSnapshot(buttonComponent);
  });
});
