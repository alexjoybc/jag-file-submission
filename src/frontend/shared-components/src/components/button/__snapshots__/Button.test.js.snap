// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button matches the snapshot 1`] = `
<div>
  <button
    className="normal-blue btn"
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  >
    Button label
  </button>
</div>
`;
